src/client/features/interactions/index.js

```diff
@@ -0,0 +1,129 @@
+const React = require('react');
+const h = require('react-hyperscript');
+const _ = require('lodash');
+const queryString = require('query-string');
+const Loader = require('react-loader');
+
+const make_cytoscape = require('../../common/cy/');
+const interactionsStylesheet= require('../../common/cy/interactions-stylesheet');
+const { ServerAPI } = require('../../services/');
+
+const { BaseNetworkView } = require('../../common/components');
+const { getLayoutConfig } = require('../../common/cy/layout');
+
+class Interactions extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cy: make_cytoscape({ headless: true, stylesheet: interactionsStylesheet }),
+      componentConfig: {},
+      layoutConfig: {},
+      networkJSON: {},
+      networkMetadata: {
+        name: '',
+        datasource: '',
+        comments: []
+      },
+      loading: true,
+    };    
+    const query = queryString.parse(props.location.search);
+    ServerAPI.getNeighborhood(query.ID).then(res=>{ 
+      const layoutConfig = getLayoutConfig('interactions');
+      const componentConfig = _.merge({}, BaseNetworkView.config, { useSearchBar: true});
+      const network= this.parse(res,query.ID);
+      this.setState({
+        componentConfig: componentConfig,
+        layoutConfig: layoutConfig,
+        networkJSON: network ,
+        networkMetadata: {
+          name: query.ID,
+          datasource: 'Pathway Commons',
+          comments: ['place holder'], dataSource: ['place holder'], organism: ['place holder'], title: ['place holder'], uri: 'place holder'
+        },
+        loading: false
+      }); 
+    });
+  }
+
+  edgeType(type){
+    switch(type){
+      case 'in-complex-with':
+      case 'interacts-with':
+        return 'Binding';
+      case 'controls-phosphorylation-of':
+        return 'Phosphorylation';
+      case 'controls-expression-of':
+        return 'Expression';
+      default:
+        return '';
+    }
+  }
+
+  findId(data,id){
+    return data.filter(line => line.includes(id))[0].split('	')[0];
+  }
+
+  pathwayLinks(sources){
+    return sources.split(';').map( link => {
+      const splitLink=link.split('/').reverse();
+      return [splitLink[1]==='reactome'? 'reactome': 'Pathway Commons',splitLink[0]];
+    });
+  }
+
+  addInteraction(nodes,edge,sources,network,nodeMap){
+    for (let i = 0; i<2; i++){
+      if(!nodeMap.has(nodes[i])){
+        nodeMap.set(nodes[i],true);
+        network.nodes.push({data:{class: "ball",id: nodes[i],label: nodes[i],parsedMetadata:[]}});
+      }
+    }
+    network.edges.push({data: {
+      id: nodes[0]+edge+nodes[1] ,
+      label:nodes[0]+' '+edge+' '+nodes[1] ,
+      source:nodes[0],
+      target: nodes[1],
+      class: this.edgeType(edge),
+      parsedMetadata:[['Database IDs',sources]]
+    }});
+  }
+
+  parse(data,query){
```

Comment:
what does this parse function do?

Reply:
Takes in the fetched data, and the gene that was searched for and returns an object of the network cut to 1 degree that BaseNetworkView can display.